doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+        richTextArea.getInputMap().registerKey(KeyBinding.shortcut(KeyCode.W), RichTextArea.PASTE_PLAIN_TEXT);
+```
+
+
+#### Redefining Function Mapped to an Existing Function Tag
+
+This example illustrates how to modify a function mapped to the function tag.  The existing key mapping will be automatically use the new behavior.  The modified mapping can also be reverted back to its original behavior.
+
+For example, the application might offer an alternative way to navigate over words using the keyboard:
+
+```java
+        // redefine a function
+        richTextArea.getInputMap().registerFunction(RichTextArea.MOVE_WORD_NEXT, () -> {
+            // apply alternative navigation logic
+            TextPos p = getCustomNextWordPosition(richTextArea);
+            richTextArea.select(p);
+        });
+
+        // revert back to the default behavior
+        richTextArea.getInputMap().restoreDefaultFunction(RichTextArea.MOVE_WORD_NEXT);
+```
+
+
+#### Setting Side Decorators
+
+This example illustrates how to set a side paragraph decorator (in this case, a LineNumberDecorator):
+
+```java
+        // sets a side decorator
+        richTextArea.setLeftDecorator(new LineNumberDecorator());
+```
+
+
+#### Providing Custom Scroll Bars
+
+In this example, a custom scroll bar will be used by the RichTextArea:
+
+```java
+        // sets a custom vertical scroll bar
+        ConfigurationParameters cp = ConfigurationParameters.
+            builder().
+            verticalScrollBar(ScrollBar::new).
+            build();
+        RichTextArea richTextArea = new RichTextArea(cp, null);
+```
+
+
+
+### Extensibility
+
+While RichTextArea should be useful enough out-of-the-box, it is designed with extensibility in mind.  The application developers can extend the functionality by:
+
+- extending the model
+- adding new data format handlers, or replacing the default ones
+- adding support for new attributes to the model
+- adding new function tags and corresponding public methods 
+
+
+
+#### Extending the Model
+
+A typical use case for extending one of the standard models is to interface with a data source: a file, a stream, or some other kind of data generated on the fly.  The choice of class to be extended is determined by whether the data can be changed by the application, whether the model is allowed to be editable by the user, and whether the data will be stored by the model in memory.  Most likely, however, the application developer would need to extend the base class - StyledTextModel.
+
+
+##### View-Only Models
+
+A read-only model can be used to present rich text document not editable by the user (for example, a help or an informational page, or a virtualized view backed by a large file).
+
+**SimpleViewOnlyStyledModel** is suitable for a small in-memory styled document.  This model provides a number of methods to populate the document one segment at a time:
+
+- SimpleViewOnlyStyledModel **addSegment**(String text)
```

Comment:
the reason for view-only is that the model is mutable, i.e. it can be populated, but once installed in the control it is not intended to be editable.
another possibility is to create a builder, though I feel it might be complicating things - the (easier?) pattern is to create and populate the model, then use it in the control.
let's discuss.